tests/components/enphase_envoy/conftest.py

```diff
@@ -385,3 +314,63 @@ def mock_setup():
 def serial_number_fixture():
     """Define a serial number fixture."""
     return "1234"
+
+
+@pytest.fixture(name="mock_go_on_grid")
+def go_on_grid_fixture():
+    """Define a go_on_grid fixture."""
+    return AsyncMock(return_value="[]")
+
+
+@pytest.fixture(name="mock_go_off_grid")
+def go_off_grid_fixture():
+    """Define a go_off_grid fixture."""
+    return AsyncMock(return_value="[]")
+
+
+@pytest.fixture(name="mock_update_dry_contact")
+def update_dry_contact_fixture():
+    """Define a update_dry_contact fixture."""
+    return AsyncMock(return_value="[]")
+
+
+@pytest.fixture(name="mock_open_dry_contact")
+def open_dry_contact_fixture():
+    """Define a gopen dry contact fixture."""
+    return AsyncMock(return_value="[]")
+
+
+@pytest.fixture(name="mock_close_dry_contact")
+def close_dry_contact_fixture():
+    """Define a close dry contact fixture."""
+    return AsyncMock(return_value="[]")
+
+
+@pytest.fixture(name="mock_enable_charge_from_grid")
+def enable_charge_from_grid_fixture():
+    """Define a enable charge from grid fixture."""
+    return AsyncMock(return_value="[]")
+
+
+@pytest.fixture(name="mock_disable_charge_from_grid")
+def disable_charge_from_grid_fixture():
+    """Define a disable charge from grid fixture."""
+    return AsyncMock(return_value="[]")
+
+
+@pytest.fixture(name="mock_set_storage_mode")
+def set_storage_mode_fixture():
+    """Define a update_dry_contact fixture."""
+    return AsyncMock(return_value="[]")
+
+
+@pytest.fixture(name="mock_set_reserve_soc")
+def set_reserve_soc_fixture():
+    """Define a update_dry_contact fixture."""
+    return AsyncMock(return_value="[]")
+
```

Comment:
Its a bit unexpected that these fixtures return empty data

Reply:
Test for these use `assert_awaited_once()`, `assert_called_with()` and `reset_mock` to validate HA is internally processing correctly towards the pyenphase methods. The return values are not used afaik. Testing of the pyenphase library methods is done in the library tests.